### PR TITLE
Support call hierarchy on type signature & add plugin to generic config  & docs

### PR DESCRIPTION
**Pull request by July541**
- Call hierarchy on type signature has no response is wired indeed, it has the same result like definition. cc @Ailrun 
- A config has been added, users can switch to enable it or not.
- Call hierarchy plugin's document has a changelog field now.

Can we move the [previous](https://github.com/haskell/haskell-language-server/pull/2067) pr to this?

### CODE REVIEW

**Comment by jneira** [comment]
> Can we move the previous pr to this?

If this one is written onto the old one and it became obsolete, we can simply close the old one

File: plugins/hls-call-hierarchy-plugin/test/Main.hs [hunk 1]
```diff
@@ -164,6 +164,20 @@ prepareCallHierarchyTests =
           selRange = mkRange 1 13 1 14
           expected = mkCallHierarchyItemC "A" SkConstructor range selRange
       oneCaseWithCreate contents 1 13 expected
+  , testGroup "type singature"
```
Comment (jneira) [inline plugins/hls-call-hierarchy-plugin/test/Main.hs:167]:
```suggestion
  , testGroup "type signature"
```

small typo

Comment (July541) [reply plugins/hls-call-hierarchy-plugin/test/Main.hs:167]: 😅

File: plugins/hls-call-hierarchy-plugin/README.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+![Call Hierarchy in Emacs](call-hierarchy-in-emacs.gif)
+
+![Call Hierarchy in VSCode](call-hierarchy-in-vscode.gif)
+
+## Prerequisite
+None. You can experience the whole feature without any setting.
+
+## Configuration
+Enable default. You can disable it in the setting whenever you like.
```
Comment (jneira) [inline plugins/hls-call-hierarchy-plugin/README.md:17]:
```suggestion
Enabled by default. You can disable it in your editor settings whenever you like.

```json
{
  "haskell.plugin.callHierarchy.globalOn": true
}
```
```

Comment (jneira) [reply plugins/hls-call-hierarchy-plugin/README.md:17]: @July541 have you checked the hls commands related with the config include hierarchy (`haskell-language-server vscode-extension-schema` and `haskell-language-server generate-default-config` ? I think the generic config option to enable/disable globally it is `haskell.plugin.callHierarchy.globalOn` but not sure at 100%

**Comment by jneira** [comment]
It would be great add a link to the plugin readme in the main repo readme

File: plugins/hls-call-hierarchy-plugin/README.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
+# Call hierarchy plugin for the [Haskell Language Server](https://github.com/haskell/haskell-language-server#readme)
+
+The call hierarchy plugin can review the code to determine where functions are called and how they relate to other functions.
+
+This plugin is useful when debugging and refactoring code because it allows you to see how different parts of the code are related. And it is more conducive for users to quickly understand their macro architecture in the face of strange code.
```
Comment (jneira) [inline plugins/hls-call-hierarchy-plugin/README.md:5]: :heart: this intro

File: plugins/hls-call-hierarchy-plugin/README.md (hunk 2 above)
Comment (July541) [reply plugins/hls-call-hierarchy-plugin/README.md:17]: I tested the two commands, it's really like you said!

**Comment by July541** [comment]
@jneira I added a table list of all plugins we have currently, is this your desire?

**Comment by jneira** [comment]
> @jneira I added a table list of all plugins we have currently in the main readme, is this your desire?

The table looks great but maybe it is better to have done the plugin docs and then add the table. That table could live in the issue about (#2066) while it is unfinished and move to README later. Write the table in a comment in that issue would be fine. There are some missing plugins which live in ghcide but dont worry, we can add them after moving it in a comment of #2066.

For now i was thinking in adding a item in the list of features, somewhat like

```
* Function call hierarchy: [see demo here](/link/to/call-hierarchy/readme.md#demo)
* Many more (TBD)
```

File: hls-plugin-api/hls-plugin-api.cabal [hunk 4]
```diff
@@ -1,6 +1,6 @@
 cabal-version: 2.4
 name:          hls-plugin-api
-version:       1.2.0.0
+version:       1.2.1.0
```
Comment (pepeiborra) [inline hls-plugin-api/hls-plugin-api.cabal:3]:
```suggestion
version:       1.2.0.1
```

File: plugins/hls-call-hierarchy-plugin/hls-call-hierarchy-plugin.cabal [hunk 5]
```diff
@@ -1,6 +1,6 @@
 cabal-version:      2.4
 name:               hls-call-hierarchy-plugin
-version:            1.0.0.0
+version:            1.0.1.0
```
Comment (pepeiborra) [inline plugins/hls-call-hierarchy-plugin/hls-call-hierarchy-plugin.cabal:3]:
```suggestion
version:            1.0.0.1
```

File: plugins/hls-call-hierarchy-plugin/README.md [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+## Configuration
+Enabled by default. You can disable it in your editor settings whenever you like.
+
+```json
+{
+  "haskell.plugin.callHierarchy.globalOn": true
+}
+
+## Change log
+### 1.0.1.0
```
Comment (pepeiborra) [inline plugins/hls-call-hierarchy-plugin/README.md:25]:
```suggestion
### 1.0.0.1
```

**Review by pepeiborra — APPROVED** [review]
Very nice! Just note that the changes to both packages are "other changes" according to the [PVP](https://pvp.haskell.org) and the version bumps can be therefore smaller.

**Comment by July541** [comment]
@jneira I removed the edit of the main readme, you may insert the features when you have the final idea.

**Review by jneira — APPROVED** [review]
LGTM, thanks again!

**Review by Ailrun — APPROVED** [review]
Awesome, thank you!

**Pull request merged**